File: Server_Go/routes/activityRoutes.go
package routes

import (
	"github.com/gorilla/mux"
)

func RegisterActivityRoutes(r *mux.Router) {
	r.HandleFunc("/", activityController.getActivity).Name("Get Activity")
	r.HandleFunc("/path", activityController.getActivitiesByPath).Name("Get Activities By Path")
	r.HandleFunc("/create", activityController.createActivity).Name("Create Activitiy")
	r.HandleFunc("/update", activityController.updateActivity).Name("Update Activitiy")
}

File: Server/routes/ActivityRoutes.js
const express = require('express');
const router = express.Router();
const activitiesController = require('../controllers/activityController.js');
const isAuthenticated = require('../Utils/AuthenticationUtil.js');

router.get('/', isAuthenticated, activitiesController.getActivity);

router.get('/path', activitiesController.getActivitiesByPath);

router.post('/create', isAuthenticated, activitiesController.createActivity);

router.post('/update', isAuthenticated, activitiesController.updateActivity);

module.exports = router;
